express-post-json/server.ts

```diff
@@ -0,0 +1,44 @@
+import express from 'express';
+
+type Grade = {
+  id: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+type NextId = number;
+
+const grades: Record<number, Grade> = {};
+
+let nextId: NextId = 1;
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const localArr: Grade[] = Object.values(grades);
+  res.json(localArr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const grade = req.body;
+  grade.id = nextId;
+  grades[nextId] = grade;
+  nextId++;
+
+  if (
```

Comment:
Good error checking, but you should do that _before_ you modify `grades`. Otherwise, you'll add an illegal grade to `grades`.

Reply:
understood, I will make sure that the error is returned, so the code execution is stopped and only one response is returned.